spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/data/SparkVortexValueReaders.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.spark.data;
+
+import dev.vortex.api.Array;
+import dev.vortex.api.DType;
+import java.util.function.BiFunction;
+import org.apache.iceberg.util.DateTimeUtil;
+import org.apache.iceberg.vortex.VortexValueReader;
+import org.apache.spark.unsafe.types.UTF8String;
+
+public class SparkVortexValueReaders {
+  private SparkVortexValueReaders() {}
+
+  public static VortexValueReader<UTF8String> utf8String() {
+    return UTF8Reader.INSTANCE;
+  }
+
+  public static VortexValueReader<Integer> date(DType.TimeUnit timeUnit) {
+    return new DateReader(timeUnit);
+  }
+
+  public static VortexValueReader<Long> timestamp(DType.TimeUnit timeUnit) {
+    // Spark timestamp has µs precision
```

Comment:
Currently Parquet stores timestamps as `INT96` and it has millisecond precision
There is ongoing work for adding nanosecond precision timestamps

Reply:
That is true but I believe what Spark expects us to return is Microseond? https://spark.apache.org/docs/3.5.3/api/java/org/apache/spark/sql/types/TimestampType.html

Reply:
Here is how it is done in Parquet (ParquetUtil.java):
```
  /**
   * Method to read timestamp (parquet Int96) from bytebuffer. Read 12 bytes in byteBuffer: 8 bytes
   * (time of day nanos) + 4 bytes(julianDay)
   */
  public static long extractTimestampInt96(ByteBuffer buffer) {
    // 8 bytes (time of day nanos)
    long timeOfDayNanos = buffer.getLong();
    // 4 bytes(julianDay)
    int julianDay = buffer.getInt();
    return TimeUnit.DAYS.toMicros(julianDay - UNIX_EPOCH_JULIAN)
        + TimeUnit.NANOSECONDS.toMicros(timeOfDayNanos);
  }
```

Reply:
So I made a mistake in my comment `millisecond` -> `microsecond`
Sorry